src/app/organization/sms-campaigns/common-resolvers/campaign.resolver.ts

```diff
@@ -0,0 +1,31 @@
+/** Angular Imports */
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+
+/** rxjs Imports */
+import { Observable } from 'rxjs';
+
+/** Custom Services */
+import { OrganizationService } from 'app/organization/organization.service';
+
+/**
+ * SMS Campaign data resolver.
+ */
+@Injectable()
+export class CampaignResolver implements Resolve<Object> {
+
+  /**
+   * @param {OrganizationService} organizationService Organization service.
+   */
+  constructor(private organizationService: OrganizationService) {}
+
+  /**
+   * Returns the smscampaign data.
+   * @returns {Observable<any>}
+   */
+  resolve(route: ActivatedRouteSnapshot): Observable<any> {
+    const smscampaignId  = route.parent.paramMap.get('id');
```

Comment:
```suggestion
    const smsCampaignId  = route.parent.paramMap.get('id');

Reply:
also make the required changes in the rest of this file and in the organization service.